gpu_direct_rdma_access.c

```diff
@@ -838,6 +881,189 @@ struct rdma_device *rdma_open_device_server(struct sockaddr *addr)
     return NULL;
 }
 
+//===========================================================================================
+static
+int rdma_exec_task(struct rdma_exec_params *exec_params) 
```

Comment:
I think this is an overkill for reusing this part of the code
It's correct, just an overkill.

Isn't it simple to have a small helper to submit the MARKER? using all dummy values which will fail... QP is already in ERROR :)

```
static int submit_marker_helper() {
    qpex.wr_id = WR_ID_FLUSH_MARKER;
    qpex.flags = SIGNAL;
    ibv_wr_start(qpex);
    ibv_wr_rdma_rw_post(qpex, 0 /*rkey*/, NULL /*remote_addr*/);
    ibv_wr_set_sge(qpex, 0 /*lkey*/, NULL /*local_addr*/, 0 /*length*/);
    mlx5dv_wr_set_dc_addr(qpex, NULL /*ah*/, 0 /*remote_dctn*/, 0 /*remote_dc_key*/);
    ibv_wr_complete(qpex);
}
```

Reply:
I checked that for successful execution we need to provide `ah, qpex, mpex `.
pros of this way:
-  code cleanup was made
-  less duplication fragments now exists 
-  we can reuse the same method, it means that logic less coupled.